content/articles/javafx-editable-table-view/index.md

```diff
@@ -0,0 +1,442 @@
+  button:
+
+![Project dependencies](new-javafx-editable-tableview-project-dependencies.png "Project dependencies")
+
+#### Initial Project Structure
+
+The application generated has the following structure:
+
+```shell
+.
+├── src
+│   ├── main
+│       ├── java
+│          ├── com.table.editabletableview
+│                 ├── HelloApplication.java
+│                 └── HelloController.java
+│          └── module-info.java
+│       └── resources
+│          └── com.table.editabletableview
+│             └── hello-view.fxml
+├── editable-tableview.iml
+└── pom.xml
+```
+
+#### New Project Structure
+
+- Create a new folder in the `java/com/table/editabletableview` path and name it `model`. It will hold the dummy data.
+- Create another folder named `controller` in the same path.
+- Change the location of the `HelloController.java` file to the new 'controller' folder created. This can be simply done by right-clicking on the file and pasting it to the new location.
+- In the 'model' folder, create a new file named `User.java`.
+
+The resulting folder structure looks as follows:
+
+```shell
+.
+├── src
+│   ├── main
+│       ├── java
+│          ├── com.table.editabletableview
+│               ├── controller
+│                   └── HelloController.java
+│               └── model
+│                   └── User.java
+│               └── HelloApplication.java
+│          └── module-info.java
+│       └── resources
+│          └── com.table.editabletableview
+│             └── hello-view.fxml
+├── editable-tableview.iml
+└── pom.xml
+```
+
+### Set up the project, tools, and development environment
+
+To achieve this, do the following:
+
+- Open the FXML file found in the `resources/com/table/editabletableview` path.
+- The IDE will produce a prompt notification at the top of the file to install JavaFX SDK. Just allow it and the IDE
+  will automatically set it up and configure it for the project.
+- On the bottom left area near the status bar click on the 'SceneBuilder' option to view and develop the application using the SceneBuilder.
+- If it is the first time to create a JavaFX project using the IDE on the machine, it will give prompt notification to install the SceneBuilder. Just click on the 'Install' option and the IDE will download it and set it up.
+- Restart the IDE if it does not automatically load to the SceneBuilder view.
+
+### Table creation and design
+
+- In the SceneBuilder view, delete the 'VBox' holding the 'Hello' button by simply right-clicking on it and selecting the Delete option. The code below will be deleted. This can be viewed in the 'Text' view.
+
+```xml
+
+<VBox alignment="CENTER" spacing="20.0" xmlns:fx="http://javafx.com/fxml"
+      fx:controller="com.table.editabletableview.HelloController">
+    <padding>
+        <Insets bottom="20.0" left="20.0" right="20.0" top="20.0"/>
+    </padding>
+
+    <Label fx:id="welcomeText"/>
+    <Button text="Hello!" onAction="#onHelloButtonClick"/>
+</VBox>
+```
```

Comment:
Please explain this code snippet using bulleted points.


Make sure to do this for all the code snippets

Reply:
@srishilesh, this code is auto-filled by the editor or IDEA

Reply:
@srishilesh I dont think there is much to explain here.

Reply:
To those which can be turned, I have and to the others I've just left them